compiler-rt/lib/msan/msan_flags.inc

```diff
@@ -22,6 +22,7 @@ MSAN_FLAG(int, origin_history_size, Origin::kMaxDepth, "")
 MSAN_FLAG(int, origin_history_per_stack_limit, 20000, "")
 MSAN_FLAG(bool, poison_heap_with_zeroes, false, "")
 MSAN_FLAG(bool, poison_stack_with_zeroes, false, "")
+MSAN_FLAG(bool, poison_in_calloc, true, "")
```

Comment:
do we really need a separate flag for this? we don't have `poison_in_posix_memalign` either.

Reply:
*everything* shares this flag (malloc, realloc, valloc, aligned_alloc, memalign, posix_memalign), which means it means "poison in alloc" essentially. IMO adding a new flag adds more confusion, not less.

Reply:
> which means it means "poison in alloc" essentially

But that's not what the flag is named.

Besides, there's a difference in operation:
- `poison_in_malloc` poisons the entire requested allocation plus the padding
- `poison_in_calloc` only poisons the padding

Reply:
That's what the flag does though; it doesn't matter what it is called. If we started adding separate flags for all the other operations, and didn't respect the `poison_in_malloc` flag for them anymore, that would break people.

Leaving to @vitalybuka to tie-break here. If we are doing this, call it something else. `poison_in_malloc` + `poison_in_calloc` implies there are `poison_in_X`, which there aren't; and there probably never will be.

Reply:
> Besides, there's a difference in operation:

(If you called it `poison_padding` and respected that flag both in `malloc` and `calloc`, I'd have no problem)